### PR TITLE
Add ability to show warning when a certain fraction of data equals a …

### LINKED ISSUE #243: Check for zeros in satellite images
## Detailed Description
Check for zeros in satellite images, if there are more than 10% then remove

## Context
We had a forecast spike when there was an load of zeros in the satellite image (by accident)

## Possible Implementation
- For each timeslice check for zeros, if >10% then remove?

### PR DESCRIPTION
**Pull request by jacobbieker**
…given value

# Pull Request

## Description

This adds checking for zeros or other numbers in data, and gives a warning if so.

Fixes #243 

## How Has This Been Tested?

Unit test

- [x] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: tests/validation/test_check_equality.py [hunk 1]
```diff
@@ -19,3 +20,14 @@ def test_check_less_than_or_equal_to(topo_datapipe):
...
+
+def test_check_value_equal_to_fraction_no_warning(topo_datapipe):
+    topo_datapipe = CheckValueEqualToFraction(topo_datapipe, value=0.0, fraction=0.1)
+    data = next(iter(topo_datapipe))
+
+
+def test_check_value_equal_to_fraction_with_warning(topo_datapipe):
+    topo_datapipe = CheckValueEqualToFraction(topo_datapipe, value=0.0, fraction=0.0)
+    with pytest.raises(Warning):
+        data = next(iter(topo_datapipe))
```
Comment (peterdudfield) [inline tests/validation/test_check_equality.py:33]: nice, good tests

**Review by peterdudfield — APPROVED** [review]

**Comment by peterdudfield** [comment]
Do we need  / should we also put this in the pvnet_2 pipeline? Atleast for Live?

**Review by dfulu — APPROVED** [review]

**Comment by jacobbieker** [comment]
> Do we need / should we also put this in the pvnet_2 pipeline? Atleast for Live?

Yeah, we can, I can add that

**Comment by dfulu** [comment]
I don't know if its worth doing this on this pull request, but I find the function name a little unclear. Could it be something like `CheckFractionOfFractionsEqualTo` and `CheckFractionOfFractionsGreaterThanOrEqualTo`?

**Pull request merged**